File: lab_2/task/Main.java
import java.io.FileWriter;
import java.io.IOException;
import java.security.SecureRandom;
import java.util.logging.Level;
import java.util.logging.Logger;

public class Main {
    private static final Logger logger = Logger.getLogger(Main.class.getName());

    public static void main(String[] args) {
        String file = "java_random.txt";

        try (FileWriter writer = new FileWriter(file)) {

            SecureRandom secureRandom = new SecureRandom();

            for (int i = 0; i < 128; ++i) {
                writer.write(String.valueOf(secureRandom.nextInt(2)));
            }

            logger.info("Sequence was generated and wrote in " + file);
        } catch (IOException e) {
            logger.log(Level.SEVERE, "Cant open txt to write in random sequence.", e);
        }
    }
}

File: lab_2/task/main.cpp
#include <iostream>
#include <fstream>
#include <random>

int main() {
	std::ofstream file("cpp_random.txt");

	if (!file.is_open()) {
		std::cerr << "Cant open txt to write in random sequence.\n";
		return 1;
	}

	std::random_device rd;
	std::mt19937 gen(rd());
	std::uniform_int_distribution<> dis(0, 1);

	for (int i = 0; i < 128; ++i) {
		file << dis(gen);
	}

	file.close();
	std::cout << "Sequence was generated and writed in cpp_random.txt" << std::endl;

	return 0;
}
